src/librustc_trans/debuginfo/metadata.rs

```diff
@@ -1517,22 +1520,66 @@ enum EnumDiscriminantInfo {
     NoDiscriminant
 }
 
+//There's no good place to put this.
+//It is used here and in prepare_enum_metadata only.
+fn discriminant_to_ty<'a, 'tcx>(tcx: &ty::TyCtxt<'a, 'tcx, 'tcx>,
```

Comment:
Why?  The other function is disappearing, and it's only used in that file.  I don't see any reason that other code should wish to go from a `Integer` to a `Ty`.  If there is a good reason to do this for anything but debuginfo, I will, but it seems to me that such a conversion being widespread is bad design.